src/MaterialDesignControls/ControlsMaterial3/MaterialEntry.cs

```diff
@@ -0,0 +1,453 @@
+
+        public TextTransforms TextTransform
+        {
+            get { return (TextTransforms)GetValue(TextTransformProperty); }
+            set { SetValue(TextTransformProperty, value); }
+        }
+
+        public static readonly BindableProperty ReturnTypeProperty =
+            BindableProperty.Create(nameof(ReturnType), typeof(ReturnType), typeof(MaterialEntry), defaultValue: ReturnType.Default);
+
+        public ReturnType ReturnType
+        {
+            get { return (ReturnType)GetValue(ReturnTypeProperty); }
+            set { SetValue(ReturnTypeProperty, value); }
+        }
+
+        public static readonly BindableProperty ReturnCommandProperty =
+            BindableProperty.Create(nameof(ReturnCommand), typeof(ICommand), typeof(MaterialEntry), defaultValue: null);
+
+        public ICommand ReturnCommand
+        {
+            get { return (ICommand)GetValue(ReturnCommandProperty); }
+            set { SetValue(ReturnCommandProperty, value); }
+        }
+
+        public static readonly BindableProperty ReturnCommandParameterProperty =
+            BindableProperty.Create(nameof(ReturnCommandParameter), typeof(object), typeof(MaterialEntry), defaultValue: null);
+
+        public object ReturnCommandParameter
+        {
+            get { return (object)GetValue(ReturnCommandParameterProperty); }
+            set { SetValue(ReturnCommandParameterProperty, value); }
+        }
+
+        public static readonly BindableProperty TextProperty =
+            BindableProperty.Create(nameof(Text), typeof(string), typeof(MaterialEntry), defaultValue: null, propertyChanged: OnTextChanged, defaultBindingMode: BindingMode.TwoWay);
+
+        public string Text
+        {
+            get { return (string)GetValue(TextProperty); }
+            set { SetValue(TextProperty, value); }
+        }
+
+        public static readonly BindableProperty MaxLengthProperty =
+            BindableProperty.Create(nameof(MaxLength), typeof(int), typeof(MaterialEntry), defaultValue: Int32.MaxValue);
+
+        public int MaxLength
+        {
+            get { return (int)GetValue(MaxLengthProperty); }
+            set { SetValue(MaxLengthProperty, value); }
+        }
+
+        public static readonly BindableProperty CursorPositionProperty =
+            BindableProperty.Create(nameof(CursorPosition), typeof(int), typeof(MaterialEntry), defaultValue: 0);
+
+        public int CursorPosition
+        {
+            get { return (int)GetValue(CursorPositionProperty); }
+            set { SetValue(CursorPositionProperty, value); }
+        }
+
+        #endregion Properties
+
+        #region Events
+
+        public event EventHandler TextChanged;
+
+        public new event EventHandler<FocusEventArgs> Focused;
+
+        public new event EventHandler<FocusEventArgs> Unfocused;
+
+        public static readonly BindableProperty FocusedCommandProperty =
```

Comment:
Can we move the FocusedCommand and UnfocusedCommand to the base class? From my point of view, all controls that will use the base class will have focus and unfocus features.